src/components/HomePage/SearchResultsArea/Offer/OfferInteractionOptions.js

```diff
@@ -51,6 +52,7 @@ const OfferInteractionOptions = ({
     const handleApplyURLRedirect = useCallback(() => {
         setClosedRedirectDialog();
         window.open(offer.applyURL, "_blank", "noopener");
+        if (offer) recordApplyURLVisit(offer._id, offer.title, offer.ownerName);
```

Comment:
is the `if` needed?

Reply:
I'm not completely sure, but I added it as a safeguard since similar verifications are done across this file to the `offer` variable
![image](https://user-images.githubusercontent.com/28673635/197515070-43455be1-fc67-4669-946f-09728be06dc6.png)

Reply:
Well, it doesn't hurt but I think the program will crash in line 54 if the offer is null. The nullable offer should be handled in line 78 and never be null. We should maybe instead have something like `if (!offer) return;` on the top of the function

Reply:
@monkin77 address this so we can merge this PR.